app/services/store_sp_metadata_in_session.rb

```diff
@@ -0,0 +1,27 @@
+class StoreSpMetadataInSession
```

Comment:
This is not the first introduction of this pattern. See CreateVerifiedAccountEvent, RequestPasswordReset, UpdateUser, and UpdateUserPassword.

Micropurchase does this as well.

I think it's fine to have some classes be verbs and others nouns, depending on the best way to describe them.